packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -859,12 +862,397 @@ class ResizeImageKey {
     }
     return other is ResizeImageKey
         && other._providerCacheKey == _providerCacheKey
+        && other._policy == _policy
         && other._width == _width
-        && other._height == _height;
+        && other._height == _height
+        && other._allowUpscaling == _allowUpscaling;
   }
 
   @override
-  int get hashCode => Object.hash(_providerCacheKey, _width, _height);
+  int get hashCode => Object.hash(_providerCacheKey, _policy, _width, _height, _allowUpscaling);
+}
+
+/// Configures the behavior for [ResizeImage].
+///
+/// This is used in [ResizeImage.policy] to affect how the [ResizeImage.width]
+/// and [ResizeImage.height] properties are interpreted.
+enum ResizeImagePolicy {
+  /// Sizes the image to the exact width and height specified by
+  /// [ResizeImage.width] and [ResizeImage.height].
+  ///
+  /// If [ResizeImage.width] and [ResizeImage.height] are both non-null, the
+  /// output image will have the specified width and height, stretching the
+  /// aspect ratio of the image if necessary. If only one of `width` and
+  /// `height` is non-null, then the output image will be scaled to the
+  /// associated width or height, and the other dimension will take whatever
+  /// value is needed to maintain the image's original aspect ratio.
+  ///
+  /// If [ResizeImage.allowUpscaling] is false, the width and the height of the
+  /// output image will each be clamped to the intrinsic width and height of the
+  /// image.
```

Comment:
nit: 

> This may result in a different aspect ratio than the aspect ratio specified by the target width and height, if for example the height gets clamped downwards but the width does not.

Reply:
Also, consider mentioning that this is similar to `[BoxFit.fill]` with consideration for null values on width and height, so setting one of those to null would end up corresponding to `[BoxFit.fitWidth]` or `[BoxFit.fitHeight]`.

Consider mentioning that the `fit` member is similar to `[BoxFit.contain]`.

I'm not sure if that would confuse more people or just lead to asking why we don't use BoxFit here, but to me having that linkage helps.